pkg/server.go

```diff
@@ -57,19 +56,32 @@ func Start(cfg *Config) {
 		timeout = pluginConfig.Timeout
 	}
 
+	logrusLevel, err := logrus.ParseLevel(cfg.LogLevel)
+	if err != nil {
+		logrus.WithError(err).Fatal("unable to set the log level")
+		logrusLevel = logrus.ErrorLevel
+	}
+
 	httpServer := &http.Server{
-		Handler:      loggedRouter,
+		Handler:      router,
 		Addr:         fmt.Sprintf(":%d", cfg.Port),
 		TLSConfig:    tlsConfig,
 		ReadTimeout:  timeout,
 		WriteTimeout: timeout,
 	}
 
+	if logrusLevel == logrus.TraceLevel {
+		loggedRouter := handlers.LoggingHandler(log.Logger.Out, router)
+		httpServer.Handler = loggedRouter
+	}
+
```

Comment:
```suggestion

logrus.SetLevel(logrusLevel)
```

Reply:
This wasn't done in the `monitoring-plugin` on purpose to add in the final log.Info of which port and http(s) the server is listening on. IMO the server start up logs should be excluded from the log filtering, as it provides pertinent information on the behavior of the server, and not just a generic log

Reply:
Restructured to output the port before setting the log level in latest commit. 

```
	log.Infof("listening on https://:%d", cfg.Port)
	logrus.SetLevel(logrusLevel)
```

Reply:
This solution doesn't work as it will incorrectly state that a http server is listening on https

Reply:
Could we just make it generic then and say 'listening on port :%d"

Reply:
I'm also okay just leaving it as it was, what do you guys think?

```
	if cfg.CertFile != "" && cfg.PrivateKeyFile != "" {
		log.Infof("listening on https://:%d", cfg.Port)
		logrus.SetLevel(logrusLevel)
		panic(httpServer.ListenAndServeTLS(cfg.CertFile, cfg.PrivateKeyFile))
	} else {
		log.Infof("listening on http://:%d", cfg.Port)
		logrus.SetLevel(logrusLevel)
		panic(httpServer.ListenAndServe())
	}
```

Reply:
Without knowing if it is https or http, a user wouldn't be able to make a successful request as the inverse will fail. I think it would be best to leave it showing https or http, it has helped me a good deal while debuging

Reply:
Ok, updated in latest commit to revert it back to have conditional print outs for `log.Infof("listening on https://:%d", cfg.Port)` and `log.Infof("listening on http://:%d", cfg.Port)`.